### ISSUE TITLE
[BUG] archlint flags NestJS lifecycle methods (onModuleInit, onModuleDestroy) as dead symbols

### ISSUE THREAD

**Issue by ewaly** [archlinter/archlint#30]
**Describe the bug**
The `archlint` tool incorrectly identifies NestJS framework lifecycle methods (e.g., `onModuleInit`, `onModuleDestroy`) as "Dead Symbols". These methods are automatically invoked by the NestJS framework and do not require explicit call expressions within the application code.

**To Reproduce**

1. Create a NestJS service/provider and implement the `OnModuleDestroy` interface.
2. Define the `onModuleDestroy()` method to handle cleanup logic (e.g., closing database connections).
3. Run `archlint scan`.
4. Observe that `onModuleDestroy` is flagged as a Dead Symbol with high severity.

**Expected behavior**
Framework-specific lifecycle methods should be recognized as implicitly used or excluded from the "Dead Symbol" analysis by default when common frameworks (like NestJS) are detected.

**Screenshots/Output**
```text
│ 🟠 HIGH  ┆ Dead Symbol                                       ┆ src/database/vectra-bootstrap.service.ts:83:9 (Class Method 'VectraBootstrapService.onModuleDestroy' definition)                             ┆ 50    │
│          ┆ (VectraBootstrapService.onModuleDestroy)          ┆                                                                                                                                              ┆       │
```

**Desktop (please complete the following information):**
- OS: macOS (Darwin 24.6.0)
- Version: 15.6
- archlint version: 0.15.0

**Additional context**
The project uses the NestJS framework. Classes are typically decorated with `@Injectable()` and implement standard interfaces from `@nestjs/common`. Since `archlint` likely looks for explicit call sites, it misses the fact that these methods are called via reflection/internals by the framework itself.
